ddtrace/contrib/flask/monkey.py

```diff
@@ -0,0 +1,487 @@
+
+    collect_view_args=True,
+    distributed_tracing_enabled=False,
+    template_default_name='<memory>',
+    trace_signals=True,
+    error_codes=set([500, ]),
+))
+
+
+# Extract flask version into a tuple e.g. (0, 12, 1) or (1, 0, 2)
+# DEV: This makes it so we can do `if flask_version >= (0, 12, 0):`
+# DEV: Example tests:
+#      (0, 10, 0) >= (0, 10)
+#      (0, 10, 0) >= (0, 10, 0)
+#      (0, 10, 1) >= (0, 10)
+#      (0, 11, 1) >= (0, 10)
+#      (0, 11, 1) >= (0, 10, 2)
+#      (1, 0, 0) >= (0, 10)
+#      (0, 9) == (0, 9)
+#      (0, 9, 0) != (0, 9)
+#      (0, 8, 5) <= (0, 9)
+flask_version_str = getattr(flask, '__version__', '0.0.0')
+flask_version = tuple([int(i) for i in flask_version_str.split('.')])
+
+
+def patch():
+    """
+    Patch `flask` module for tracing
+    """
+    # Check to see if we have patched Flask yet or not
+    if getattr(flask, '_datadog_patch', False):
+        return
+    setattr(flask, '_datadog_patch', True)
+
+    # Attach service pin to `flask.app.Flask`
+    Pin(
+        service=config.flask['service_name'],
+        app=config.flask['app'],
+        app_type=config.flask['app_type'],
+    ).onto(flask.Flask)
+
+    # flask.app.Flask methods that have custom tracing (add metadata, wrap functions, etc)
+    _w('flask', 'Flask.wsgi_app', traced_wsgi_app)
+    _w('flask', 'Flask.dispatch_request', traced_dispatch_request)
+    _w('flask', 'Flask.add_url_rule', traced_add_url_rule)
+    _w('flask', 'Flask.endpoint', traced_endpoint)
+    _w('flask', 'Flask._register_error_handler', traced_register_error_handler)
+
+    # flask.blueprints.Blueprint methods that have custom tracing (add metadata, wrap functions, etc)
+    _w('flask', 'Blueprint.register', traced_blueprint_register)
+    _w('flask', 'Blueprint.add_url_rule', traced_blueprint_add_url_rule)
+
+    # flask.app.Flask traced hook decorators
+    flask_hooks = [
+        'before_request',
+        'before_first_request',
+        'after_request',
+        'teardown_request',
+        'teardown_appcontext',
+    ]
+    for hook in flask_hooks:
+        _w('flask', 'Flask.{}'.format(hook), traced_flask_hook)
+    _w('flask', 'after_this_request', traced_flask_hook)
+
+    # flask.app.Flask traced methods
+    flask_app_traces = [
+        'preprocess_request',
+        'process_response',
+        'handle_exception',
+        'handle_http_exception',
+        'handle_user_exception',
+        'try_trigger_before_first_request_functions',
+        'do_teardown_request',
+        'do_teardown_appcontext',
+        'send_static_file',
+    ]
+    for name in flask_app_traces:
+        _w('flask', 'Flask.{}'.format(name), simple_tracer('flask.{}'.format(name)))
+
+    # flask static file helpers
+    _w('flask', 'send_file', simple_tracer('flask.send_file'))
+    # Skipping for now since it basically does `return send_file(join(dir, filename))`
+    # _w('flask', 'send_from_directory', simple_tracer('flask.send_from_directory'))
```

Comment:
probably, my thinking was to leave it in in case someone was going through to improve this, to know why we skipped over it... but.. just adds noise to a huge file.